### PR TITLE
[Core] Added module override support to webpack

### PR DESCRIPTION
**Pull request by ridz1208**
This pull request does what the title says it does.

To test:
- create the project config file (see below for example)
- add specific jsx to compile in the same manner they are added in the loris webpack config
- run `npm run compile` make sure both the loris and project files are compiled properly

### CODE REVIEW

**Comment by ridz1208** [comment]
@driusan given that alot of the overridden modules in projects now use JSX files and need to be transpiled... this would be very nice in 17.1 for projects to be able to maintain overrides

**Comment by ridz1208** [comment]
@jacobpenny feel free to suggest any changes or best practices

**Comment by johnsaigle** [comment]
@ridz1208 Travis is mad

File: webpack.config.js [hunk 1]
```diff
@@ -87,4 +88,11 @@ var config = {
   plugins: [new webpack.optimize.UglifyJsPlugin({mangle: false})]
 };
 
+// Support project overrides
+if (fs.existsSync('./project/webpack-project.config.js')) {
+  exists=true;
```
Comment (jacobpenny) [inline webpack.config.js]: Is this used somewhere?

File: webpack.config.js [hunk 2]
```diff
@@ -87,4 +88,11 @@ var config = {
...
+  projConfig = require('./project/webpack-project.config.js');
```
Comment (jacobpenny) [inline webpack.config.js]: Should add a `var` to limit the scope of this variable.
```var projConfig = require('./project/webpack-project.config.js');```

**Comment by ridz1208** [comment]
@jacobpenny please re-review

**Comment by ridz1208** [comment]
Example study config file:
```
var webpack = require('webpack');
var path = require('path');

var config = {
  entry: {
    './project/modules/candidate_parameters/js/index.js': './project/modules/candidate_parameters/jsx/index.js'
  },
  output: {
    path: './',
    filename: '[name]'
  },
  module: {
    loaders: [
      {
        test: /\.js$/,
        exclude: /node_modules/,
        loader: 'babel-loader'
      }
    ]
  },
  resolve: {
    root: path.resolve(__dirname)+'/../',
    alias: {
      util: 'htdocs/js/util',
      jsx: 'jsx',
      Breadcrumbs: 'jsx/Breadcrumbs',
      DynamicDataTable: 'jsx/DynamicDataTable',
      FilterForm: 'jsx/FilterForm',
      Form: 'jsx/Form',
      Markdown: 'jsx/Markdown',
      PaginationLinks: 'jsx/PaginationLinks',
      Panel: 'jsx/Panel',
      ProgressBar: 'jsx/ProgressBar',
      StaticDataTable: 'jsx/StaticDataTable',
      Tabs: 'jsx/Tabs'
    },
    extensions: ['', '.js', '.jsx']
  },
  externals: {
    react: 'React'
  },
  devtool: 'source-map',
  plugins: [new webpack.optimize.UglifyJsPlugin({mangle: false})]
};

module.exports = config;
```

**Comment by ridz1208** [comment]
@ZainVirani you should be able to test this one, no rush for now

File: webpack.config.js [hunk 3]
```diff
@@ -87,4 +88,10 @@ var config = {
   plugins: [new webpack.optimize.UglifyJsPlugin({mangle: false})]
 };
 
+// Support project overrides
+if (fs.existsSync('./project/webpack-project.config.js')) {
+  var projConfig = require('./project/webpack-project.config.js');
+  config = [config, projConfig];
```
Comment (ZainVirani) [inline webpack.config.js]: This doesn't work when I tested it with CAP's `project/webpack-project.config.js`
It does work with `config.push(projConfig);` instead

Comment (ridz1208) [reply webpack.config.js]: screenshot from 2017-10-24 15-59-52

Comment (ZainVirani) [reply webpack.config.js]: weird -- works for me

Comment (ridz1208) [reply webpack.config.js]: @ZainVirani it would work if you config is an array
the one currently on loris is not, are you sure urs is not modified?

Comment (ZainVirani) [reply webpack.config.js]: you're right -- disregard my comments

**Comment by ridz1208** [comment]
@ZainVirani I incorporated all your changes and mine.

please test and review

**Comment by ZainVirani** [comment]
tested on CAP's set up and a normal Loris setup, works fine

**Pull request merged**